src/pybind/ctc/ctc_pybind.cc

```diff
@@ -0,0 +1,215 @@
+
+  // we use the default CUDA stream by setting it to nullptr, i.e., 0
+  //
+  // As far as PyTorch is concerned, we can get the underlying CUDA stream in
+  // PyTorch (the `_cdata` attribute) and pass it to kaldi pybind. But this
+  // approach is too tricky and is specific to PyTorch.
+  //
+  // FYI (fangjun): the relevant information to implement the above approach is
+  // here
+  //
+  // https://github.com/pytorch/pytorch/blob/master/torch/csrc/cuda/Module.cpp#L117
+  // https://github.com/pytorch/pytorch/blob/master/torch/cuda/__init__.py#L431
+  // https://github.com/pytorch/pytorch/blob/master/c10/core/Stream.h#L104
+  // https://github.com/pytorch/pytorch/blob/master/c10/core/Stream.h#L126
+}
+
+}  // namespace
+
+void pybind_ctc(py::module& kaldi_m) {
+  py::module m = kaldi_m.def_submodule("ctc", "pybind for warp-ctc");
+
+  m.def("GetWarpCtcVersion", &get_warpctc_version,
+        "Returns a single integer which specifies the API version of the "
+        "warpctc library");
+
+  py::enum_<ctcStatus_t>(m, "CtcStatus", py::arithmetic())
+      .value("CTC_STATUS_SUCCESS", CTC_STATUS_SUCCESS)
+      .value("CTC_STATUS_MEMOPS_FAILED", CTC_STATUS_MEMOPS_FAILED)
+      .value("CTC_STATUS_INVALID_VALUE", CTC_STATUS_INVALID_VALUE)
+      .value("CTC_STATUS_EXECUTION_FAILED", CTC_STATUS_EXECUTION_FAILED)
+      .value("CTC_STATUS_UNKNOWN_ERROR", CTC_STATUS_UNKNOWN_ERROR);
+
+  m.def(
+      "CtcGetStatusString", &ctcGetStatusString,
+      "Returns a string containing a description of status that was passed in",
+      py::arg("status"));
+
+  py::enum_<ctcComputeLocation>(m, "CtcComputeLocation", py::arithmetic())
+      .value("CTC_CPU", CTC_CPU)
+      .value("CTC_GPU", CTC_GPU);
+
+  py::class_<ctcOptions>(m, "CtcOptions",
+                         "Structure used for options to the CTC computation. "
+                         "Applications should zero out the array using memset "
+                         "and sizeof(struct ctcOptions) in C or default "
+                         "initialization (e.g. 'ctcOptions options{};' or "
+                         "'auto options = ctcOptions{}') in C++ to ensure "
+                         "forward compatibility with added options.")
+      .def(py::init<>())
+      .def_readwrite("loc", &ctcOptions::loc,
+                     "indicates where the ctc calculation should take place "
+                     "{CTC_CPU | CTC_GPU}")
+      .def_readwrite("num_threads", &ctcOptions::num_threads,
+                     "used when loc == CTC_GPU, which stream the kernels "
+                     "should be launched in")
+      .def_readwrite("blank_label", &ctcOptions::blank_label,
+                     "the label value/index that the CTC calculation should "
+                     "use as the blank label");
+
+  const char* compute_ctc_loss_help_str =
+      "Compute the connectionist temporal classification loss between a "
+      "sequence of probabilities and a ground truth labeling on **CPU**.  "
```

Comment:
This help string is being used for both CPU and GPU code but says it's for CPU.

Reply:
fixed.